cpp/src/arrow/type_traits.h

```diff
@@ -901,6 +918,34 @@ static inline bool is_floating(Type::type type_id) {
   return false;
 }
 
+/// \brief Check for a numeric type
+///
```

Comment:
Since we are at it, let's make the docstrings even more explicit
```suggestion
///
/// This predicate doesn't match decimals (see `is_decimal`).
```